ui/src/dropdown.tsx

```diff
@@ -152,119 +169,160 @@ const
+      onChecked = (itemName: S) => (_ev?: React.FormEvent<HTMLElement | HTMLInputElement>, checked = false) => {
+        setItems(items => items.map(item => (({...item, checked: item.name === itemName ? checked : false, show: true}))))
+        wave.args[name] = itemName
+        model.value = itemName
+        if (trigger) wave.push()
         toggleDialog()
-        setFilteredItems(items.map(i => { i.checked = initialSelectedMap.has(i.name); return i }))
-      }, [initialSelectedMap, items, toggleDialog]),
-      submit = React.useCallback((checkedItem?: DropdownItem) => {
-        const result = checkedItem ? [checkedItem] : items.filter(({ checked }) => checked)
-        wave.args[name] = result.length === 1 ? result[0].name : result.map(({ name }) => name)
-
+      }
+    
+      React.useEffect(() => {
+        if (model.value !== undefined) {
+          wave.args[name] = model.value ?? null
+          setItems(items => items.map(i => ({ ...i, checked: model.value === i.name })))
+        }
+      }, [name, model.value, setItems])
+    
+      return (
+        <>
+          <Fluent.TextField
+            data-test={name}
+            iconProps={{ iconName: 'ChevronDown' }}
+            readOnly
+            onClick={toggleDialog}
+            label={label}
+            disabled={disabled}
+            required={required}
+            styles={{ field: { cursor: 'pointer' }, icon: { fontSize: 12, color: cssVar('$neutralSecondary') } }}
+            placeholder={placeholder}
+            value={items.find(i => i.checked)?.text ?? ''} />
+          <Fluent.Dialog hidden={isDialogHidden} dialogContentProps={{ title: label, type: Fluent.DialogType.close }} onDismiss={toggleDialog} minWidth={600} maxWidth='90vw'>
+            <Fluent.DialogContent styles={{ innerContent: { height: '65vh' }, header: { height: 0 } }}>
+              <Fluent.Label>Search</Fluent.Label>
+              <Fluent.SearchBox data-test={`${name}-search`} onChange={onSearchChange} autoFocus />
+              <Fluent.ScrollablePane styles={{ root: { marginTop: 100, boxShadow: `0px 3px 7px ${cssVar('$text3')}` } }}>
+                <Fluent.List
+                  items={items.filter(i => i.show)}
+                  getPageSpecification={getPageSpecification}
+                  renderedWindowsAhead={3}
+                  onRenderCell={onRenderCell(onChecked)}
+                />
+              </Fluent.ScrollablePane>
+            </Fluent.DialogContent>
+          </Fluent.Dialog>
+        </>
+      )
+  },
+  DialogDropdownMulti = ({ name, choices = [], values = [], disabled, required, trigger, placeholder, label }: Dropdown) => {
+    const
+      [isDialogHidden, setIsDialogHidden] = React.useState(true),
+      [items, setItems, onSearchChange] = useItems(choices, values),
+      auxItems = React.useRef([...items]),
+      openDialog = () => {
+        setIsDialogHidden(false)
+        auxItems.current = [...items]
```

Comment:
Same question as in initialization.

Reply:
[Discussion here ](https://github.com/h2oai/wave/pull/1621#discussion_r979763603)